src/osd.c

```diff
@@ -364,6 +378,23 @@ render_osd(struct server *server, cairo_t *cairo, int w, int h,
 			case LAB_FIELD_TYPE:
 				buf_add(&buf, get_type(*view));
 				break;
+			case LAB_FIELD_WINFO:
+				buf_add(&buf, (*view)->workspace->name);
```

Comment:
For the workspace name I think it makes sense to guard it by the `all_workspaces` setting, otherwise its just useless space wasted by duplicating the same workspace name. We could also check if there is more than one workspace defined (the `show_workspace` bool could be reused for that I guess).

Reply:
I put an if all workspaces around workspace name, but replaced with 4 spaces as an else.
But now that I've done it, I'm not sure about that. 
What do you think?

Reply:
I don't think the 4 spaces are necessary. Suggest something like this:
```c
if (rc.window_switcher.all_workspaces && show_workspace) {
	buf_add(&buf, (*view)->workspace->name);
}
```
This should ensure that the workspace name only occupies space when its actually required (e.g. no space wasted when the switcher is only showing the windows from the current workspace or when there is just one workspace defined anyway).

Reply:
Alright, should be good to go